src/meta-srv/src/ddl.rs

```diff
@@ -0,0 +1,155 @@
+            kv_store: self.kv_store.clone(),
+            datanode_clients: self.datanode_clients.clone(),
+            notifier: None,
+        }
+    }
+}
+
+impl DdlContext {
+    pub fn with_notifier(&mut self, notifier: oneshot::Sender<ProcedureStatus>) {
+        self.notifier = Some(notifier)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum ProcedureStatus {
+    CreateTable(CreateTableProcedureStatus),
+}
+
+impl DdlManager {
+    pub(crate) fn new(
+        procedure_manager: ProcedureManagerRef,
+        selector: SelectorRef,
+        selector_ctx: SelectorContext,
+        table_id_sequence: SequenceRef,
+        kv_store: KvStoreRef,
+        datanode_clients: Arc<DatanodeClients>,
+    ) -> Self {
+        Self {
+            procedure_manager,
+            selector,
+            selector_ctx,
+            table_id_sequence,
+            kv_store,
+            datanode_clients,
+        }
+    }
+
+    pub(crate) fn create_context(&self) -> DdlContext {
+        DdlContext {
+            selector: self.selector.clone(),
+            selector_ctx: self.selector_ctx.clone(),
+            table_id_sequence: self.table_id_sequence.clone(),
+            kv_store: self.kv_store.clone(),
+            datanode_clients: self.datanode_clients.clone(),
+            notifier: None,
+        }
+    }
+
+    pub(crate) fn try_start(&self) -> Result<()> {
+        let context = self.create_context();
+
+        self.procedure_manager
+            .register_loader(
+                CreateTableProcedure::TYPE_NAME,
+                Box::new(move |json| {
+                    let context = context.clone();
+                    CreateTableProcedure::from_json(json, context).map(|p| Box::new(p) as _)
+                }),
+            )
+            .context(error::RegisterProcedureLoaderSnafu {
+                type_name: CreateTableProcedure::TYPE_NAME,
+            })
+    }
+
+    pub(crate) async fn execute_procedure_task(
+        &self,
+        cluster_id: u64,
+        task: DdlTask,
+    ) -> Result<(ProcedureId, ProcedureStatus)> {
+        let (tx, mut rx) = oneshot::channel();
+
+        let procedure_with_id = match task {
+            DdlTask::CreateTable(create_table_task) => {
+                let mut context = self.create_context();
+                context.with_notifier(tx);
+
+                let procedure = CreateTableProcedure::new(cluster_id, create_table_task, context);
+                ProcedureWithId::with_random_id(Box::new(procedure))
+            }
+        };
+
+        let procedure_id = procedure_with_id.id;
+
+        let mut watcher = self
+            .procedure_manager
+            .submit(procedure_with_id)
+            .await
+            .context(error::SubmitProcedureSnafu)?;
+
+        watcher::wait(&mut watcher)
+            .await
+            .context(error::WaitProcedureSnafu)?;
+
+        let status = rx.try_recv().context(error::TryReceiveStatusSnafu)?;
```

Comment:
Isn't `watcher::wait` not sufficient to indicate the create table procedure is done?

Reply:
We need a channel to receive the response from the procedure.(e.g., created table_id, etc. )